src/amo/components/Collection/index.js

```diff
@@ -197,10 +214,22 @@ export class CollectionBase extends React.Component<Props> {
   }
 }
 
-export const mapStateToProps = (state: { collections: CollectionsState }) => {
+export const mapStateToProps = (
+  state: {| collections: CollectionsState, user: UserStateType |}
+) => {
+  const { current: collection, loading } = state.collections;
+
+  let hasEditPermission = false;
+
+  if (collection) {
+    hasEditPermission = collection.authorId === state.user.id ||
+      hasPermission(state, COLLECTIONS_EDIT);
```

Comment:
Should this be outside of the `if (collection)` block so that admins always get permission rights? But maybe if `collection` is falsy it doesn't matter anyway?

Reply:
mmh if the collection is not loaded, there is no link. Maybe admins could get a link *before* but I am not sure that is matters a lot.